### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by SUPERCILEX**
12!

### CODE REVIEW

File: constants.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ project.ext {
     targetSdk = 27
     minSdk = 14
 
-    firebaseVersion = '11.8.0'
+    firebaseVersion = '12.0.0'
```
Comment (samtstern) [inline constants.gradle]: Let's wait for `12.0.1` ... `12.0.0` turned out to be kinda broken by some wayward ProGuard.

Comment (SUPERCILEX) [reply constants.gradle]: Yeah, I just saw your other comment. SGTM!

Comment (SUPERCILEX) [reply constants.gradle]: @samtstern Can you also unmark the `CredentialsClient` as `@Hide`? That makes lint think we're not allowed to use its methods... pretty sure you guys only meant to mark the constructors as `@Hide` 😄.

**Comment by samtstern** [comment]
That's another 12.0.0 bug, it's getting fixed.

**Comment by SUPERCILEX** [comment]
Awesome sauce!

**Comment by samtstern** [comment]
@SUPERCILEX 12.0.1 is out, so we should be able to update this, merge it, and release `3.3.0`

**Comment by samtstern** [comment]
@SUPERCILEX looks like `:app:lint` is not happy?  But I think that's my fault, I will merge this and fix that myself.

**Comment by SUPERCILEX** [comment]
Sweet, I was just about to tell you I don't have my laptop with me, thanks! ❤️

**Pull request merged**